plugins/jobs/src/js/data/JobModel.ts

```diff
@@ -0,0 +1,293 @@
+import { makeExecutableSchema, IResolvers } from "graphql-tools";
+import { Observable } from "rxjs";
+
+export interface Query {
+  jobs: JobConnection | null;
+  job: Job | null;
+}
+
+export interface GeneralArgs {
+  [key: string]: any;
+}
+
+export interface JobsQueryArgs {
+  filter?: string | null;
+  sortBy?: SortOption | null;
+  sortDirection?: SortDirection | null;
+}
+
+export interface JobQueryArgs {
+  id: string;
+  filter?: string | null;
+  sortBy?: SortOption | null;
+  sortDirection?: SortDirection | null;
+}
+
+export type SortOption = "ID" | "STATUS" | "LAST_RUN";
+
+export type SortDirection = "ASC" | "DESC";
+
+export interface JobConnection {
+  filteredCount: number;
+  totalCount: number;
+  nodes: Job[];
+}
+
+export interface Job {
```

Comment:
I'd rather see these declarations split in several files: https://github.com/dcos/dcos-ui/pull/3007/files
I think your declarations are much more precise though

Reply:
The idea behgínd having it here is that a) they are generated from the GraphQL schema and b) that we can use them (import) for any of the other files. What would speak for moving them away from their source?

Reply:
Aren't we going to have duplications that way?

Reply:
Why should we have duplicates this way?

Reply:
Correct me if I'm wrong. We have js files `Job.js` `JobRun.js` etc that we have to write `.d.ts` for. Plus we have these declarations that are the same? If they aren't than it is way too confusing and we have to figure out a way making them less ambiguous.

Reply:
this `Job` is something different to the `Job` struct… eventually these will replace the struct entirely.

Reply:
Also: Keep in mind that this file isnt in use yet, once it is, `Job` struct will be gone (i know, you're writing all the definitions right now, but thats part of the downside of parallelization :()

Reply:
OK this is really confusing. But let's see how it turns out